iothub/device/src/Transport/AmqpIoT/AmqpIoTMessageConverter.cs

```diff
@@ -218,84 +258,78 @@ public static void UpdateAmqpMessageHeadersAndProperties(AmqpMessage amqpMessage
-                    {
-                        amqpObject = ReadStream((Stream)netObject);
-                    }
-                    break;
-                case PropertyValueType.Uri:
-                    amqpObject = new DescribedType((AmqpSymbol)UriName, ((Uri)netObject).AbsoluteUri);
-                    break;
-                case PropertyValueType.DateTimeOffset:
-                    amqpObject = new DescribedType((AmqpSymbol)DateTimeOffsetName, ((DateTimeOffset)netObject).UtcTicks);
-                    break;
-                case PropertyValueType.TimeSpan:
-                    amqpObject = new DescribedType((AmqpSymbol)TimeSpanName, ((TimeSpan)netObject).Ticks);
-                    break;
-                case PropertyValueType.Unknown:
-                    if (netObject is Stream)
-                    {
-                        if (mappingType == MappingType.ApplicationProperty)
-                        {
-                            amqpObject = ReadStream((Stream)netObject);
-                        }
-                    }
-                    else if (mappingType == MappingType.ApplicationProperty)
-                    {
-                        throw FxTrace.Exception.AsError(new SerializationException(IotHubApiResources.GetString(ApiResources.FailedToSerializeUnsupportedType, netObject.GetType().FullName)));
-                    }
-                    else if (netObject is byte[])
-                    {
-                        amqpObject = new ArraySegment<byte>((byte[])netObject);
-                    }
-                    else if (netObject is IList)
-                    {
-                        // Array is also an IList
-                        amqpObject = netObject;
-                    }
-                    else if (netObject is IDictionary)
-                    {
-                        amqpObject = new AmqpMap((IDictionary)netObject);
-                    }
-                    break;
-                default:
-                    break;
+                throw Fx.Exception.ArgumentNull("amqpMessage");
             }
 
-            return amqpObject != null;
+            string methodRequestId = string.Empty;
+            string methodName = string.Empty;
+
+            SectionFlag sections = amqpMessage.Sections;
+            if ((sections & SectionFlag.Properties) != 0)
+            {
+                // Extract only the Properties that we support
+                methodRequestId = amqpMessage.Properties.CorrelationId != null ? amqpMessage.Properties.CorrelationId.ToString() : null;
+            }
+
+            if ((sections & SectionFlag.ApplicationProperties) != 0)
+            {
+                if (!(amqpMessage.ApplicationProperties?.Map.TryGetValue(new MapKey(MethodName), out methodName) ?? false))
+                {
+                    Fx.Exception.TraceHandled(new InvalidDataException("Method name is missing"), "MethodConverter.ConstructMethodRequestFromAmqpMessage");
```

Comment:
We shouldn't throw `InvalidDataException` unless we were throwing this before.

Reply:
We were throwing this before - this is exactly the old converter code.

---
In reply to: [292153912](https://github.com/Azure/azure-iot-sdk-csharp/pull/947#discussion_r292153912) [](ancestors = 292153912)

Reply:
In that case let's leave it as-is. (Not ideal.)